cmd/server-main.go

```diff
@@ -255,6 +256,9 @@ func serverMain(ctx *cli.Context) {
 	if ctx.Args().First() == "help" || !endpointsPresent(ctx) {
 		cli.ShowCommandHelpAndExit(ctx, "server", 1)
 	}
+	runtime.MemProfileRate = 4096      // 512K -> 4K - Must be constant throughout application lifetime.
```

Comment:
Probably we can do this in `func init()` ?

Reply:
We *could*, but I'm not a big fan of import sideeffects. 

Is there any other commands we need to cover? You can't profile in "gateway" mode as far as I know.

Reply:
Ah yes the side effects, the reason I thought was no one imports minio/minio/cmd - since this is internal to us not as a library.

Reply:
Probably we can move this in a separate function

Reply:
@vadmeste Yeah, I am more a fan of that. 

Having an `init` modify global state is bad practice, even if you assume that you will never be imported.

I am having a bit hard seeing what problem we are solving by moving it from here, though.